test/integration/uniswap-v3-add-support-gas.spec.ts

```diff
@@ -0,0 +1,69 @@
+import hre, { deployments, ethers } from 'hardhat';
+import { evm, wallet } from '@utils';
+import { getNodeUrl } from 'utils/env';
+import { SignerWithAddress } from '@nomiclabs/hardhat-ethers/signers';
+import { UniswapV3Adapter } from '@typechained';
+import { constants, utils } from 'ethers';
+import { DeterministicFactory, DeterministicFactory__factory } from '@mean-finance/deterministic-factory/typechained';
+import { setTestChainId } from 'utils/deploy';
+import { expect } from 'chai';
+import { given, then, when } from '@utils/bdd';
+import { TransactionResponse } from '@ethersproject/providers';
+import { snapshot } from '@utils/evm';
+
+const CHAIN = { chain: 'optimism', chainId: 10 };
+const BLOCK_NUMBER = 13838662;
+const BYTES = '0xf2c047db4a7cf81f935c'; // Some random bytes
+
+const DAI = '0xda10009cbd5d07dd0cecc66161fc93d7c9000da1';
+const RAI = '0x7fb688ccf682d58f86d7e38e03f9d22e7705448b';
+
+describe('Uniswap v3 Add Support - Gas Test', () => {
+  let deployer: SignerWithAddress;
+  let oracle: UniswapV3Adapter;
+  let snapshotId: string;
+
+  before(async () => {
+    const { deployer: deployerAddress } = await hre.getNamedAccounts();
+    deployer = await ethers.getSigner(deployerAddress);
+    await fork({ ...CHAIN, blockNumber: BLOCK_NUMBER });
+    await deployments.fixture(['UniswapV3Adapter'], { keepExistingDeployments: false });
```

Comment:
Had to set `keepExistingDeployments` to `false` make it work, not sure why. If I don't do this, then the tx doesn't revert... Weird...